util/multirun.sh

```diff
@@ -23,9 +23,41 @@ set -o pipefail
 
 usage()
 {
-    echo >&2 "usage: $0 <path/to/cip/binary> [<path/to/manifest.yaml>,<path/to/service-account.json>, ...]"
-    echo >&2 "The 2nd argument onwards are '<manifest>,<service-account>,<service-account>...' strings."
-    echo >&2
+cat <<EOF >&2
+usage: $0 <path/to/cip/binary> [<MANIFEST>,<KEYFILE>[,<KEYFILE>,...] ...]
+
+MANIFEST: path/to/manifest.yaml
+KEYFILE: path/to/service-account.json
+
+The 2nd argument onwards are
+
+    '<manifest>,<service-account>,<service-account>...'
+
+strings.
+
+If an execution does not require any service account, pass in '<manifest>,'
+(notice the trailing comma).
+
+If you only want to run against manifests that changed in a Git commitish,
+specify the CIP_GIT_REV and CIP_GIT_DIR environment variables. E.g.,
+CIP_GIT_REV=HEAD
```

Comment:
Hmm. That's a good point. I didn't think of this scenario for some reason --- I could easily just force the user to specify a range of commits. I need to cross-check the Pod Utilities helper in the Prow job to see if that information is even available for this script at runtime.

Reply:
According to the [Pod Utilities docs](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md), it *merges* the PR into the base branch. So we lose information about "where is HEAD in upstream?" (in order to get a list of commits in the PR by diffing commit-wise against origin/master vs the PR).

The workaround is to just add the upstream github repo as a remote and do a fetch (then just diff against remote); then we can diff against origin/master and local HEAD to get a list of all commits in the PR. A bit finicky, but doable with a handful of shell. The one downside is that the promoter manifest repo must not be a private repo, but I don't think that will be a concern.

Let me add that bit of logic in here...

Reply:
Or you could cache the md5sum of all the files you have synced in an emptyDir.  It doesn't even need to be super persistent, maybe - if you get moved to a new machine you do a full sync, and then start tracking changes again.  Restarts on the same machine would still be cached.  Or you could use a small volume to keep the cache.

Reply:
That sounds too complicated. I'd like to try out the idea of diffing commits first to see where that leads.

I'd like to add that the bit about no support for private promoter manifest repos is not entirely true because technically the overall Prow bot still has read access (by definition) to the manifest repo (it's just that I am not aware of the way to inject this secret into the Prow job itself). Alternatively, I can add a patch to Prow to make it define an env var that answers the question "what are the commits in the PR?", or "at what commit is upstream HEAD?".